### PR TITLE
Show bonus in list by click detail-button from the map

### PR DESCRIPTION
**Pull request by Tamila-Bodnar**

### CODE REVIEW

File: src/app/core/components/bonus-list/bonus-list.component.html [hunk 1]
```diff
@@ -1,3 +1,8 @@
+<app-bonus-detail
+  *ngIf="bonusFromMap"
+  [bonus]="bonusFromMap"
+></app-bonus-detail>
+
 <div *ngFor="let bonus of bonuses" class="bonus-list">
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list/bonus-list.component.html:6]: add trackBy to for loop

File: src/app/core/components/bonus-list/bonus-list.component.ts [hunk 2]
```diff
@@ -25,4 +29,10 @@ export class BonusListComponent implements OnInit {
       (err) => console.log(`error ${err}`)
     );
   }
+
+  public getBonusMap(): void {
+    this.bonusesService.bonusFromMap.subscribe((bonus: IBonus) => {
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list/bonus-list.component.ts:34]: you forgot to unsubscribe, check all places please

File: src/app/core/components/bonus-list/bonus-list.component.ts [hunk 3]
```diff
@@ -25,4 +29,10 @@ export class BonusListComponent implements OnInit {
       (err) => console.log(`error ${err}`)
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list/bonus-list.component.ts:29]: we have toaster service, you can use it to show error

File: src/app/core/components/bonus-list/bonus-list.component.ts [hunk 4]
```diff
@@ -9,10 +10,13 @@ import { BonusesService } from '../../services/bonuses.service';
 })
 export class BonusListComponent implements OnInit {
   public bonuses: IBonus[] = [];
+  public bonusFromMap: IBonus;
+
   constructor(public bonusesService: BonusesService) {}
 
   ngOnInit(): void {
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list/bonus-list.component.ts:17]: modificators

File: src/app/core/components/bonus-list/bonus-list.component.ts [hunk 5]
```diff
@@ -9,10 +10,13 @@ import { BonusesService } from '../../services/bonuses.service';
 })
 export class BonusListComponent implements OnInit {
   public bonuses: IBonus[] = [];
+  public bonusFromMap: IBonus;
+
   constructor(public bonusesService: BonusesService) {}
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list/bonus-list.component.ts:15]: why don't you do this service injection and requests in a container component?

File: src/app/core/components/map/bonus-popup/bonus-popup.component.ts [hunk 6]
```diff
@@ -1,20 +1,26 @@
...
   templateUrl: './bonus-popup.component.html',
-  styleUrls: ['./bonus-popup.component.scss']
+  styleUrls: ['./bonus-popup.component.scss'],
 })
 export class BonusPopupComponent implements OnInit {
-
+  constructor(
+    private bonusesService: BonusesService
+  ) {}
   @Input() public bonus: IBonus;
```
Comment (rmurtazin) [inline src/app/core/components/map/bonus-popup/bonus-popup.component.ts:15]: you need to place all variables declaration over the constructor

File: src/app/core/components/map/bonus-popup/bonus-popup.component.ts [hunk 7]
```diff
@@ -1,20 +1,26 @@
...
 
   public ngOnInit(): void {}
```
Comment (rmurtazin) [inline src/app/core/components/map/bonus-popup/bonus-popup.component.ts:17]: remove if you don't it

File: src/app/core/services/bonuses.service.ts [hunk 8]
```diff
@@ -1,37 +1,42 @@
...
 import { IBonus } from '@interfaces/bonus.interface';
-import { Observable } from 'rxjs';
+import { Observable, Subject } from 'rxjs';
 import { ApiService } from '@services/api.service';
 
 @Injectable()
-export class BonusesService{
-    constructor(private api: ApiService){}
+export class BonusesService {
+  public bonusFromMap = new Subject();
```
Comment (rmurtazin) [inline src/app/core/services/bonuses.service.ts:8]: type of property
set private modificatory

File: src/app/core/components/bonus-list/bonus-list.component.html (hunk 1 above)
Comment (rmurtazin) [reply src/app/core/components/bonus-list/bonus-list.component.html:6]: Cool🙂

File: src/app/core/components/bonus-list-view/bonus-list-view.component.html [hunk 9]
```diff
@@ -1,3 +1,3 @@
-<div *ngFor="let bonus of bonuses" class="bonus-list">
+<div *ngFor="let bonus of bonuses; trackBy: trackById" class="bonus-list">
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list-view/bonus-list-view.component.html:1]: Great 😉

File: src/app/core/components/bonus-list-view/bonus-list-view.component.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  selector: 'app-bonus-list-view',
+  templateUrl: './bonus-list-view.component.html',
+  styleUrls: ['./bonus-list-view.component.scss'],
+})
+export class BonusListViewComponent implements OnInit {
+  @Input() bonusMap: IBonus;
+  @Input() bonuses: IBonus[];
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (rmurtazin) [inline src/app/core/components/bonus-list-view/bonus-list-view.component.ts:14]: Remove this method if you don't use

**Review by rmurtazin — APPROVED** [review]

File: src/app/core/components/bonus-list-view/bonus-list-view.component.ts (hunk 10 above)
Comment (Tamila-Bodnar) [reply src/app/core/components/bonus-list-view/bonus-list-view.component.ts:14]: Thanks for all comments

**Pull request merged**